tiledb/sm/storage_manager/open_array.cc

```diff
@@ -82,15 +80,10 @@ void OpenArray::cnt_incr() {
   ++cnt_;
 }
 
-bool OpenArray::is_empty(uint64_t snapshot) const {
+bool OpenArray::is_empty(uint64_t timestamp) const {
   std::lock_guard<std::mutex> lck(mtx_);
-  for (uint64_t i = 0; i < fragment_metadata_.size(); ++i) {
-    if (snapshot < i)
-      break;
-
-    if (!fragment_metadata_[i].empty())
-      return false;
-  }
+  for (const auto& metadata : fragment_metadata_)
+    return metadata->timestamp() > timestamp;
```

Comment:
This will only check the first fragment and immediately return.

Reply:
This is actually correct. It is a more concise way to write:
```
if(fragment_metadata_.empty())
  return true;
return fragment_metadata_.front() > timestamp;
```

Note that the `for (const auto& metadata : fragment_metadata_)` iterates over the metadata ordered in ascending timestamp/UUID order (set `fragment_metadata_` uses a custom comparator).

I will change it to the above though for more clarity.

Reply:
Ah ha, yes I did not understand that, but what you say makes sense.